src/secrets.js

```diff
@@ -81,12 +95,57 @@ async function selectData(data, selector) {
     }
 
     if (result.startsWith(`"`)) {
-        result = JSON.parse(result);
+        // we need to strip the beginning and ending quotes otherwise it will
+        // always successfully parse as a JSON string
+        result = result.substring(1, result.length - 1);
+        if (!isJSONString(result)) {
+            // add the quotes back so we can parse it into a Javascript object
+            // to allow support for multi-line secrets. See https://github.com/hashicorp/vault-action/issues/160
+            result = `"${result}"`
+            result = JSON.parse(result);
+        }
+    } else if (isJSONString(result)) {
+        if (storedAsJSONData) {
+            // Support secrets stored in Vault as pure JSON.
```

Comment:
Wanted to confirm since I'm a bit unclear on when we would be getting into this block of code. `isJSONString` would tell us that the `typeof result == string`, however `storedAsJSONData` would tell us that `typeof result == object`. Is this because of an overlap in JS that a `string` is a subset of the type `object`?

Reply:
No, the JS type `string` is not a subset of type `object`. In JS strings hold [primitive](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#primitive_values) values which are immutable. [Objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Data_structures#objects) hold mutable values.

`isJSONString`  and `storedAsJSONData` are telling us different things about how the value is stored in Vault. We hit this block of code when a value is stored in Vault as "pure JSON" which vault stores as a map:


```
$ vault kv get -mount=secret  test-json-data
...

jsonData    map[x:1 y:qux]
```

vs a JSON string which Vault stores as a string:
```
$ vault kv get -mount=secret  test-json-string
...

jsonString    {"x":1,"y":"qux"}
```

Reply:
Ah I see thanks for clarifying, that makes sense now!